public/js/cin.js

```diff
@@ -0,0 +1,527 @@
+    if (user.role.name == "Interviewer") {
+      names.push(user.user.name);
+    }
+  });
+  return names.join(", ");
+};
+
+const getLatestODVideo = () => {
+  // return "7lUz0xU5d9g";
+  for (let i = interviews.length - 1; i >= 0; i--) {
+    if (interviews[i].interview.youtube_id != null) {
+      return interviews[i].interview.youtube_id;
+    }
+  }
+  return defaultYouTube; // URY Ad (there's nothing else to show :()
+};
+
+function escapeRegExp(string) {
+  return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+}
+
+const ScheduleArea = () => {
+  const [slots, setSlots] = useState([]);
+  const searchTerm = useRef("");
+  const [searched, setSearched] = useState(true);
+  const [youtubeVid, setYoutubeVid] = useState(getLatestODVideo());
+  const [youtubeTitle, setYoutubeTitle] = useState("You Just Missed");
+
+  const handleSearch = (event) => {
+    searchTerm.current = event.target.value;
+    setSearched(false);
+  };
+
+  const updateYoutube = (id) => {
+    setYoutubeVid(id);
+    setYoutubeTitle("");
+  };
+
+  const updateSchedule = (auto) => {
+    if (!auto || searchTerm.current == "") {
+      console.log("Update Schedule");
+      var tmp = [
+        html`<input
+          type="search"
+          id="search"
+          class="form-control mx-auto bg-cin"
+          placeholder="Search"
+          aria-label="Search"
+          onKeyUp=${handleSearch}
+          style="width: 25em;  max-width: 90%;"
+        />`,
+      ];
+      interviews.forEach((event) => {
+        // Spaces seem to break the search, so just yeet the space characters
+        if (
+          searchTerm.current == "" ||
+          event.interview.position.full_name
+            .toLowerCase()
+            .replace(/\s/g, "")
+            .search(
+              escapeRegExp(searchTerm.current).toLowerCase().replace(/\s/g, "")
+            ) != -1 ||
+          prettifyCandidates(event.interview.candidates)
+            .toLowerCase()
+            .replace(/\s/g, "")
+            .search(
+              escapeRegExp(searchTerm.current).toLowerCase().replace(/\s/g, "")
+            ) != -1
+        ) {
+          if (new Date(event.end_time).getTime() < Date.now()) {
+            var youtube = event.interview.youtube_id;
+            tmp.push(html`<${PastScheduleCard}
+              position=${event.interview.position.full_name}
+              candidate=${prettifyCandidates(event.interview.candidates)}
+              interviewer=${getInterviewers(event)}
+              youtubeStatus=${youtube != null
+                ? html`<a class="cin-text-2" href="javascript:void(0)"
+                    >Watch on YouTube</a
+                  >`
+                : "Available on YouTube Soon"}
+              callback=${updateYoutube}
+              youtubeID=${youtube}
+            />`);
+          } else if (new Date(event.start_time).getTime() > Date.now()) {
+            let time =
+              new Date(event.start_time).toLocaleTimeString().slice(0, -3) +
```

Comment:
Please don't rely on `toLocaleTimeString` doing sane things when you `slice()` it. Use `date-fns` `format()` instead. (Or if you're reluctant to add yet another library, which I get, consider `Intl.DateTimeFormat` with a fallback.

(What is this `slice(0, -3)` even there for anyway?)

Reply:
takes out 2 digits of seconds and 1 colon

Reply:
Not guaranteed to work - toLocaleString can vary its output depending on the locale the browser is set to. Better to use DateTimeFormat